src/cephadm/cephadm.py

```diff
@@ -3452,6 +3457,8 @@ def list_daemons(ctx, detail=True, legacy_dir=None):
                 for j in os.listdir(os.path.join(data_dir, i)):
                     if '.' in j and os.path.isdir(os.path.join(data_dir, fsid, j)):
                         name = j
+                        if daemon_name and name != daemon_name:
```

Comment:
This is fine for now, but it seems to me the whole list_daemons thing is crying out for a nice refactoring ;-)